front/lib/labs/temporal/activities.ts

```diff
@@ -0,0 +1,303 @@
+import { LabsTranscriptsHistoryResource } from "@app/lib/resources/labs_transcripts_history_resource";
+import mainLogger from "@app/logger/logger";
+
+const {
+  NANGO_GOOGLE_DRIVE_CONNECTOR_ID,
+  LABS_API_KEY,
+  LABS_WORKSPACE_ID,
+  LABS_TRANSCRIPTS_ASSISTANT,
+  NODE_ENV,
+  SENDGRID_API_KEY,
+} = process.env;
+
+export async function retrieveNewTranscriptsActivity(
+  userId: number,
+  providerId: LabsTranscriptsProviderType
+) {
+  const logger = mainLogger.child({ userId });
+
+  if (providerId == "google_drive") {
+    const auth = await getGoogleAuth(userId);
+
+    // Only pull transcripts from last day
+    // We could do from the last 15 minutes
+    // but we want to avoid missing any if the workflow is down
+    const cutoffDate = new Date();
+    cutoffDate.setDate(cutoffDate.getDate() - 1);
+
+    const files = await googleapis.google
+      .drive({ version: "v3", auth })
+      .files.list({
+        q:
+          "name contains '- Transcript' and createdTime > '" +
+          cutoffDate.toISOString() +
+          "'",
+        fields: "files(id, name)",
+      });
+
+    if (!files.data.files) {
+      logger.info("[retrieveNewTranscripts] No new files found");
+      return;
+    }
+
+    files.data.files.forEach(async (file: drive_v3.Schema$File) => {
+      const fileId = file.id as string;
+
+      // Check in history if we already processed this file
+      const history = await LabsTranscriptsHistoryResource.findByFileId({
+        fileId,
+      });
+      if (history) {
+        logger.info(
+          "[retrieveNewTranscripts] File already processed. Skipping.",
+          { fileId }
+        );
+        return;
+      }
+
+      await launchProcessTranscriptWorkflow({ userId, fileId });
+    });
+  }
+
+  logger.info(
+    {},
+    "[retrieveNewTranscripts] Successful run retrieveNewTranscriptsActivity"
+  );
+}
+
+export async function processGoogleDriveTranscriptActivity(
+  userId: number,
+  fileId: string
+) {
+  const logger = mainLogger.child({ userId });
+  const providerId = "google_drive";
+
+  const user = await User.findByPk(userId);
+  if (!user) {
+    logger.error(
+      "[processGoogleDriveTranscriptActivity] User not found. Stopping."
+    );
+    return;
+  }
+
+  if (!SENDGRID_API_KEY) {
+    throw new Error("Missing SENDGRID_API_KEY env variable");
+  }
+  sgMail.setApiKey(SENDGRID_API_KEY);
+
+  if (!NANGO_GOOGLE_DRIVE_CONNECTOR_ID) {
+    throw new Error("NANGO_GOOGLE_DRIVE_CONNECTOR_ID is not set");
+  }
+  logger.info(
+    "[processGoogleDriveTranscriptActivity] Starting processing of file ",
+    fileId
+  );
+
+  const transcriptsConfiguration =
+    await LabsTranscriptsConfigurationResource.findByUserIdAndProvider({
+      attributes: ["id", "connectionId", "provider", "agentConfigurationId"],
+      where: {
+        userId: userId,
+        provider: providerId,
+      },
+    });
+
+  if (!transcriptsConfiguration) {
```

Comment:
No, not expected - that's just a safeguard in case things change or an edge case happens. What's our policy on this? Should I just remove if because not expected?